src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -0,0 +1,667 @@
+
+	const char *commandString = renameCommand->data;
+
+	Node *parseTree = ParseTreeNode(commandString);
+
+	CitusProcessUtility(parseTree, commandString, PROCESS_UTILITY_TOPLEVEL,
+						NULL, None_Receiver, NULL);
+}
+
+
+/*
+ * RenameShardRelationConstraints appends given shardId to the end of the name
+ * of constraints "defined on" the relation with shardRelationId. This function
+ * utilizes GetConstraintNameList to pick the constraints to be renamed,
+ * see more details in function's comment.
+ */
+static void
+RenameShardRelationConstraints(Oid shardRelationId, uint64 shardId)
+{
+	List *constraintNameList = GetConstraintNameList(shardRelationId);
+
+	char *constraintName = NULL;
+	foreach_ptr(constraintName, constraintNameList)
+	{
+		const char *commandString =
+			GetRenameShardConstraintCommand(shardRelationId, constraintName, shardId);
+
+		Node *parseTree = ParseTreeNode(commandString);
+		CitusProcessUtility(parseTree, commandString, PROCESS_UTILITY_TOPLEVEL,
+							NULL, None_Receiver, NULL);
+	}
+}
+
+
+/*
+ * GetConstraintNameList returns a list of constraint names "defined on" the
+ * relation with relationId. Those constraints can be:
+ *  - "check" constraints or,
+ *  - "primary key" constraints or,
+ *  - "unique" constraints or,
+ *  - "trigger" constraints or,
+ *  - "exclusion" constraints or,
+ *  - "foreign key" constraints in which the relation is the "referencing"
+ *     relation (including the self-referencing foreign keys).
+ */
+static List *
+GetConstraintNameList(Oid relationId)
+{
+	List *constraintNameList = NIL;
+
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+
+	Relation pgConstraint = heap_open(ConstraintRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_constraint_conrelid,
+				BTEqualStrategyNumber, F_OIDEQ, relationId);
+
+	bool useIndex = true;
+	SysScanDesc scanDescriptor = systable_beginscan(pgConstraint,
+													ConstraintRelidTypidNameIndexId,
+													useIndex, NULL, scanKeyCount,
+													scanKey);
+
+	HeapTuple heapTuple = systable_getnext(scanDescriptor);
+	while (HeapTupleIsValid(heapTuple))
+	{
+		Form_pg_constraint constraintForm = (Form_pg_constraint) GETSTRUCT(heapTuple);
+
+		char *constraintName = NameStr(constraintForm->conname);
+		constraintNameList = lappend(constraintNameList, pstrdup(constraintName));
+
+		heapTuple = systable_getnext(scanDescriptor);
+	}
+
+	systable_endscan(scanDescriptor);
+	heap_close(pgConstraint, NoLock);
+
+	return constraintNameList;
+}
+
+
+/*
+ * RenameForeignConstraintsReferencingToShard appends given shardId to the end
+ * of the name of foreign key constraints in which the relation with
+ * shardRelationId is the "referenced one" except the self-referencing foreign
+ * keys. This is because, we already renamed self-referencing foreign keys via
+ * RenameShardRelationConstraints function.
+ */
+static void
+RenameForeignConstraintsReferencingToShard(Oid shardRelationId, uint64 shardId)
```

Comment:
Shouldn't this function be named `RenameForeignConstraintsReferencedByShard` or such?

Reply:
Actually here we rename constraints "referencing to shard", where the shard is the "referenced" one. The variable name  `Oid referencedTableId` was wrong, fixing it